### PR TITLE
Message transform pr feedback

### PR DESCRIPTION
**Pull request by gregschohn**
### Description
Enhancements as per feedback for [PR-147](https://github.com/opensearch-project/opensearch-migrations/pull/147). 

### Testing
All unit tests pass. `docker system prune; ./gradlew clean... :dockerSolution:composeUp`.  All tests pass along the way.

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/PacketToTransformingHttpHandlerFactory.java [hunk 1]
```diff
@@ -2,34 +2,34 @@
...
+                                                  SslContext sslContext) {
         this.jsonTransformer = jsonTransformer;
         this.eventLoopGroup = new NioEventLoopGroup();
         this.serverUri = serverUri;
         this.sslContext = sslContext;
     }
 
-    public IPacketToHttpHandler createNettyHandler() {
-        return new NettyPacketToHttpHandler(eventLoopGroup, serverUri, sslContext);
+    IPacketFinalizingConsumer createNettyHandler() {
```
[the remark this answers is not in the document]
Comment (gregschohn) [reply TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/PacketToTransformingHttpHandlerFactory.java:26]: I'm changing this to return NettyPacketToHttpConsumer

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonContentCompressor.java [hunk 2]
```diff
@@ -104,13 +97,4 @@ public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         channelUnregistered(ctx);
     }
-
-    @Override
-    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
```
[the remark this answers is not in the document]
Comment (gregschohn) [reply TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonContentCompressor.java:109]: It defers to the superclass, which is what it was doing anyway

**Review by lewijacn — APPROVED** [review]
Approved with one comment on a comment

File: TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java [hunk 3]
```diff
@@ -85,7 +85,7 @@ static class Parameters {
         @Parameter(required = true,
                 names = {"--listenPort"},
                 arity = 1,
-                description = "Port of the server that the proxy is capturing traffic for.")
```
[the remark this answers is not in the document]
Comment (gregschohn) [reply TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java:88]: Thanks!  Agree - See the latest version

**Pull request merged**